editor/scene_tree_dock.cpp

```diff
@@ -2012,17 +2012,169 @@ void SceneTreeDock::_toggle_placeholder_from_selection() {
 	}
 }
 
+void SceneTreeDock::_update_all_gizmos(Node *p_node) {
+	Node3DEditor::get_singleton()->update_all_gizmos(p_node);
+}
+
+template <class RootType>
+void SceneTreeDock::_reparent_nodes_to_typed_root(RootType *p_root, List<Node *> &p_nodes, Node *p_owner) {
+	for (Node *node : p_nodes) {
+		node->set_owner(p_owner);
+		if (node != p_root && node->get_owner() == p_owner) {
+			List<Node *> owned;
+			node->get_owned_by(p_owner, &owned, true);
+			// For Node2D and Node3D types, we can preserve the global transform when move them to the root node
+			if (typeid(RootType) == typeid(Node3D)) {
```

Comment:
typeid() may introduce RTTI.

I'd suggest a compile-time check, or using a different function for each of the three cases.

Reply:
You should be able to do `Object::cast_to<Node3D>(p_root)` to check if it's Node3D.

Reply:
It's not a huge deal, more of an attempt at good practice, but I think what I was hoping for was a means of using templates based on the root node to reduce unnessecary branching. Since my conversation with Lyuma that we determined that the way I was doing it would likely introduce RTTI, I shall attempt to change the way its done